### PR TITLE
Upgrade fc, fcoe, network set, Enclosure group, uplink set and server hardware type

### PR DESCRIPTION
**Pull request by jyothisgm**
### Description
Upgrade fc, fcoe, network set, Enclosure group, uplink set, connection template, ethernet network, enclosure and server hardware type.

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [X] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Comment by soodpr** [comment]
you can update oneview-sdk version to 5.8.0 in gemfile and metadata.json file

File: endpoints-support.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+***Notes***
+
+* If an endpoint is not marked as implemented for a specific API, it can still be used in compatibility mode for the supported API versions.
+* If an example is not working on a supported API version, verify the [HPE OneView REST API Documentation](http://h17007.www1.hpe.com/docs/enterprise/servers/oneview4.1/cic-api/en/index.html#)  for the API version being used, since the expected attributes for that resource might have changed.
+<br />
+
+## HPE OneView
+| Endpoints                                                                       | Verb     | V200 | V300 | V500 |V600 |V800 | V1000 | V1200 |
+| --------------------------------------------------------------------------------------- | -------- | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: |
```
Comment (madhav-bharadwaj) [inline endpoints-support.md]: remove versions 200, 300 and 500.

Comment (madhav-bharadwaj) [inline endpoints-support.md:19]: We dont need to include versions 200-600. Remove them.

File: metadata.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
   "requirements": [
     {"name":"puppet","version_requirement":">= 4.1.0"},
     {"name": "ruby","version_requirement": ">=2.2.3"},
-    {"name": "oneview-sdk-ruby","version_requirement": ">=5.6.0"}
+    {"name": "oneview-sdk-ruby","version_requirement": ">=5.9.0"}
```
Comment (madhav-bharadwaj) [inline metadata.json]: Is this version already released?

Comment (soodpr) [reply metadata.json]: this is the reason why build is failing..please replace it with 5.8.0

Comment (jyothisgm) [reply metadata.json]: need to release 5.9.0 from ruby then only this will work.

File: endpoints-support.md (hunk 1 above)
Comment (jyothisgm) [reply endpoints-support.md:19]: This is still there in ruby endpoint support. that's why I haven't removed them.

File: metadata.json (hunk 2 above)
Comment (soodpr) [reply metadata.json]: we can't wait for the puppet PR's pending to be merged till ruby 5.9.0 is released. Moreover we can make this change in a new PR when puppet will be released.

**Review by madhav-bharadwaj — APPROVED** [review]

**Review by soodpr — APPROVED** [review]
LGTM

**Pull request merged**